DelvUI/Interface/GeneralElements/LabelHud.cs

```diff
@@ -68,12 +68,30 @@ private void DrawLabel(string text, Vector2 parentPos, Vector2 parentSize, GameO
             }
         }
 
-        public virtual PluginConfigColor Color(GameObject? actor = null) =>
-        Config.UseJobColor switch
+        public virtual PluginConfigColor Color(GameObject? actor = null)
         {
-            true when actor is Character || actor is BattleNpc battleNpc && battleNpc.ClassJob.Id > 0 => Utils.ColorForActor(actor),
-            true when actor is not Character => GlobalColors.Instance.NPCFriendlyColor,
-            _ => Config.Color
-        };
+            switch (Config.UseJobColor)
+            {
+                case true when (actor is Character || actor is BattleNpc battleNpc && battleNpc.ClassJob.Id > 0):
+                    return Utils.ColorForActor(actor);
+                case true when actor is not Character:
+                    return GlobalColors.Instance.NPCFriendlyColor;
+            }
+
+            switch (Config.UseRoleColor)
+            {
+                case true when (actor is Character || actor is BattleNpc battleNpc && battleNpc.ClassJob.Id > 0):
+                    {
+                        Character? character = actor as Character;
```

Comment:
Are you sure this works fine if actor is a BattleNpc?
Cant remember is BattleNpc is also a Character (if this is true, the conditions are reversed)

Reply:
@Tischel had to make a fix, it works now for battleNpc